### PR TITLE
feat: added support for image editing and upscaling for imagen3

### PR DESCRIPTION
**Pull request by pavelgj**
```ts
  const baseImg = fs.readFileSync('base.png', { encoding: 'base64' });
  const maskImg = fs.readFileSync('mask.png', { encoding: 'base64' });

  const response = await ai.generate({
    model: imagen3,
    output: { format: 'media' },
    prompt: [
      { media: { url: `data:image/png;base64,${baseImg}` }},
      {
        media: { url: `data:image/png;base64,${maskImg}` },
        metadata: { type: 'mask' },
      },
      { text: 'replace the background with foo bar baz' },
    ],
    config: {
      editConfig: {
        editMode: 'outpainting',
      },
    },
  });

  return response.media();
```

### CODE REVIEW

File: js/plugins/vertexai/src/imagen.ts [hunk 1]
```diff
@@ -222,8 +258,15 @@ export function imagenModel(
...
       };
-      if (extractPromptImage(request))
-        instance.image = { bytesBase64Encoded: extractPromptImage(request)! };
+      const baseImage = extractBaseImage(request);
+      if (baseImage) {
+        instance.image = { bytesBase64Encoded: baseImage };
+      }
+      if (extractMaskImage(request)) {
+        instance.mask = {
+          image: { bytesBase64Encoded: extractBaseImage(request)! },
```
Comment (MichaelDoyle) [inline js/plugins/vertexai/src/imagen.ts:267]: Should this be a call to `extractMaskImage(...)`?

File: js/plugins/vertexai/src/imagen.ts [hunk 2]
```diff
@@ -161,10 +187,19 @@ function toParameters(
...
+    ?.media?.url.split(',')[1];
+}
+
+function extractBaseImage(request: GenerateRequest): string | undefined {
   return request.messages
     .at(-1)
-    ?.content.find((p) => !!p.media)
+    ?.content.find(
+      (p) => !!p.media && (!p.metadata?.type || p.metadata?.type === 'base')
+    )
```
Comment (MichaelDoyle) [inline js/plugins/vertexai/src/imagen.ts:202]: Is this metadata to be set by the end-user? Will we follow up w/ a docs update?

File: js/plugins/vertexai/src/imagen.ts (hunk 1 above)
Comment (pavelgj) [reply js/plugins/vertexai/src/imagen.ts]: lol, good catch. that's what I get for last minute copy-paste "fixes"

File: js/plugins/vertexai/src/imagen.ts (hunk 2 above)
Comment (pavelgj) [reply js/plugins/vertexai/src/imagen.ts:202]: yes.  added docs.

**Review by MichaelDoyle — APPROVED** [review]

**Review by Neilblaze — APPROVED** [review]

**Pull request merged**